nav2_tasks/include/nav2_tasks/is_stuck_condition.hpp

```diff
@@ -101,31 +100,30 @@ class IsStuckCondition : public BT::ConditionNode, public rclcpp::Node
       return;
     }
 
-    RCLCPP_INFO(get_logger(), msg);
+    RCLCPP_INFO(node_->get_logger(), msg);
     prev_msg = msg;
   }
 
-  void startWorkerThread()
+  void updateStates()
   {
-    spinning_ok_ = true;
-    workerThread_ = new std::thread(&IsStuckCondition::workerThread, this);
-  }
+    // Approximate acceleration
+    // TODO(orduno) #400 Smooth out velocity history for better accel approx.
+    if (odom_history_.size() > 2) {
+      auto curr_odom = odom_history_.end()[-1];
+      double t2 = static_cast<double>(curr_odom.header.stamp.sec);
```

Comment:
Might want to change `t2` to `curr_time_stamp` and `t1` to `prev_time_stamp`.

Reply:
Good idea, will do